Bucketeer/Sources/Public/BKTClient.swift

```diff
@@ -70,29 +64,38 @@ public class BKTClient {
             }
         }
     }
+    
+    private func destroy() {
+        taskScheduler?.invalidate()
+        taskScheduler = nil
+        execute { [weak self] in
+            // must destroy in the intenal queue to prevent race condition
+            self?.component.destroy()
+        }
+    }
 }
 
 extension BKTClient {
     public static func initialize(config: BKTConfig, user: BKTUser, timeoutMillis: Int64 = 5000, completion: ((BKTError?) -> Void)? = nil) throws {
         guard (Thread.isMainThread) else {
             throw BKTError.illegalState(message: "the initialize method must be called on main thread")
         }
+        guard BKTClient.default == nil else {
```

Comment:
Because we will remove the main thread requirement, we will need to check this in the concurrentQueue.

Reply:
changed https://github.com/bucketeer-io/ios-client-sdk/commit/c3cbc81e92dff55e903b98379cb980df212bf654